packages/runtime/src/widgets/Form/MultiSelect.tsx

```diff
@@ -67,9 +73,17 @@ const MultiSelect: React.FC<MultiSelectProps> = (props) => {
     },
   );
 
-  // load initial values
+  // used to store previous initial values
+  const prevInitialValue = useRef([] as string[]);
+
+  // check and load initial values
   useEffect(() => {
-    if (isArray(values?.initialValue)) {
+    // compare previous initial value with current render initial value
+    if (
+      !isEqual(prevInitialValue.current, values?.initialValue) &&
+      isArray(values?.initialValue)
+    ) {
+      prevInitialValue.current = values?.initialValue || [];
       setSelectedValues(values?.initialValue);
```

Comment:
Comments should explain why this code is here, not what it does. The code already says what it does.

This should say, `We need to deep compare the value of the binding to tell when it has changed because it's an array`